@PVince81 

````
12:01:17 PhantomJS 2.1.1 (Linux 0.0.0): Executed 835 of 838 (skipped 3) SUCCESS (20.107 secs / 19.233 secs)
12:01:18 PhantomJS 2.1.1 (Linux 0.0.0) ERROR
12:01:18   TypeError: undefined is not a function (evaluating 'this.data.state()')
12:01:18   at apps/files/js/file-upload.js:157
12:01:18 PhantomJS 2.1.1 (Linux 0.0.0): Executed 835 of 838 (skipped 3) ERROR (20.233 secs / 19.233 secs)
12:01:18 PhantomJS 2.1.1 (Linux 0.0.0) ERROR
12:01:18   TypeError: undefined is not a function (evaluating 'this.data.state()')
12:01:18   at apps/files/js/file-upload.js:157
12:01:18 PhantomJS 2.1.1 (Linux 0.0.0): Executed 835 of 838 (skipped 3) ERROR (20.234 secs / 19.233 secs)
12:01:20 [33m06 03 2017 12:01:20.039:WARN [launcher]: [39mPhantomJS was not killed in 2000 ms, sending SIGKILL.
12:01:20 Makefile:185: recipe for target 'test-js' failed
12:01:20 make: *** [test-js] Error 1

--- a/apps/files/js/file-upload.js
+++ b/apps/files/js/file-upload.js
@@ -34,6 +34,9 @@
 OC.FileUpload = function(uploader, data) {
 	this.uploader = uploader;
 	this.data = data;
+	if (!data) {
+		throw 'Missing "data" argument in OC.FileUpload constructor';
+	}
 	var path = '';
 	if (this.uploader.fileList) {
 		path = OC.joinPaths(this.uploader.fileList.getCurrentDirectory(), this.getFile().name);

--- a/apps/files/js/file-upload.js
+++ b/apps/files/js/file-upload.js
@@ -157,7 +157,13 @@ OC.FileUpload.prototype = {
 	 * @return {bool}
 	 */
 	isPending: function() {
-		return this.data.state() === 'pending';
+		if (!this.data) {
+			// this should not be possible!
+			var stack = new Error().stack;
+			console.error(stack);
+			return false;
+		}
+		return this.data && this.data.state() === 'pending';
 	},
 
 	deleteUpload: function() {

